nilearn/regions/tests/test_signal_extraction.py

```diff
@@ -9,490 +9,492 @@
+    img_to_signals_maps,
+    signals_to_img_labels,
+    signals_to_img_maps,
+)
 
-_TEST_DIM_ERROR_MSG = (
+_3D_EXPECTED_ERROR_MSG = (
     "Input data has incompatible dimensionality: "
     "Expected dimension is 3D and you provided "
     "a 4D image"
 )
+_4D_EXPECTED_ERROR_MSG = (
+    "Input data has incompatible dimensionality: "
+    "Expected dimension is 4D and you provided "
+    "a 3D image"
+)
 _TEST_SHAPE_ERROR_MSG = "Images have incompatible shapes."
 _TEST_AFFINE_ERROR_MSG = "Images have different affine matrices."
 
 INF = 1000 * np.finfo(np.float32).eps
+EPS = np.finfo(np.float64).eps
 
 
-def test__check_shape_and_affine_compatibility_without_dim():
-    """Ensure correct behaviour for valid data without dim"""
-    shape = (2, 3, 4)
-    affine = np.eye(4)
-    img1 = nibabel.Nifti1Image(np.zeros(shape), affine)
+AFFINE = np.eye(4)
 
-    img2 = nibabel.Nifti1Image(np.zeros(shape), affine)
+SHAPE = (8, 9, 10)
 
-    with pytest.warns(None):
-        signal_extraction._check_shape_and_affine_compatibility(img1, img2)
+N_TIMEPOINTS = 17
+
+N_REGIONS = 8
+
+
+def _make_label_data(shape=SHAPE):
+    labels_data = np.zeros(shape, dtype="int32")
+    h0 = shape[0] // 2
+    h1 = shape[1] // 2
+    h2 = shape[2] // 2
+    labels_data[:h0, :h1, :h2] = 1
+    labels_data[:h0, :h1, h2:] = 2
+    labels_data[:h0, h1:, :h2] = 3
+    labels_data[:h0, h1:, h2:] = 4
+    labels_data[h0:, :h1, :h2] = 5
+    labels_data[h0:, :h1, h2:] = 6
+    labels_data[h0:, h1:, :h2] = 7
+    labels_data[h0:, h1:, h2:] = 8
+    return labels_data
+
+
+@pytest.fixture
+def labels_data():
+    return _make_label_data()
+
+
+@pytest.fixture
+def labels_img():
+    labels_data = _make_label_data(SHAPE)
+    return nibabel.Nifti1Image(labels_data, AFFINE)
+
+
+@pytest.fixture
+def mask_img():
+    mask_data = np.zeros(SHAPE)
+    mask_data[1:-1, 1:-1, 1:-1] = 1
+    return nibabel.Nifti1Image(mask_data, AFFINE)
+
+
+@pytest.fixture
+def signals():
+    return generate_timeseries(n_timepoints=N_TIMEPOINTS, n_features=N_REGIONS)
 
 
-def test__check_shape_and_affine_compatibility_with_dim():
+@pytest.fixture
+def img_3D():
+    return nibabel.Nifti1Image(np.zeros(SHAPE), AFFINE)
+
+
+@pytest.fixture
+def img_4D():
+    shape = SHAPE + (7,)
+    return nibabel.Nifti1Image(np.zeros(shape), AFFINE)
+
+
+@pytest.fixture
+def fmri_img():
+    fmri_img, _ = generate_fake_fmri(shape=SHAPE, affine=AFFINE)
+    return fmri_img
+
+
+@pytest.fixture
+def labeled_regions():
+    labels = list(range(N_REGIONS + 1))  # 0 is background
+    return generate_labeled_regions(
+        shape=SHAPE, n_regions=N_REGIONS, labels=labels
+    )
+
+
+def _all_voxel_of_each_region_have_same_values(
+    data, labels_data, n_regions, signals
+):
+    for n in range(1, n_regions + 1):
+        sigs = data[labels_data == n, :]
+        np.testing.assert_almost_equal(sigs[0, :], signals[:, n - 1])
+        assert abs(sigs - sigs[0, :]).max() < EPS
+
+
+"""ERRORS"""
```

Comment:
I'd remove this

Reply:
yup
I had put some landmarks mostly to help me find my way around.

Will remove just at the end.